### PR TITLE
[GDAL] Add zstd and webp drivers

### PR DESCRIPTION
**Pull request by evetion**
Instead of upgrading, let's broaden some drivers. Had to enable -lstdc++, not sure if it's needed on other platforms than linux.

### CODE REVIEW

**Comment by evetion** [comment]
More food for https://github.com/JuliaGeo/GDAL.jl/issues/65. Failing builds are due to hash mismatches or failed downloads.

**Comment by evetion** [comment]
I believe only FreeBSD is still failing. Something something libstdc++.so.6 not found.

File: G/GDAL/build_tarballs.jl [hunk 1]
```diff
@@ -35,6 +35,7 @@ elif [[ "${target}" == *-linux-* ]]; then
     fi
     autoreconf -vi
 fi
+export LDFLAGS="$LDFLAGS -lstdc++"
```
Comment (giordano) [inline G/GDAL/build_tarballs.jl:38]: This doesn't look like a good idea, especially for macOS and FreeBSD.  Why did you need it?

Comment (evetion) [reply G/GDAL/build_tarballs.jl:38]: Thanks for checking. All linux builds failed without it, but it seems I moved it slightly to high, will move it into the linux check.

**Comment by evetion** [comment]
Btw, thanks for keeping a very close eye on the failing builds, restarting where necessary. 🥇

File: G/GDAL/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply G/GDAL/build_tarballs.jl]: Ok, I see we are already using it for other libraries when building for linux platforms, this is a bit borked

Comment (evetion) [reply G/GDAL/build_tarballs.jl]: Ok, this caused FreeBSD to fail, now works :)

Comment (giordano) [reply G/GDAL/build_tarballs.jl]: Yeah, we don't use GCC for macOS and FreeBSD by default, hence no libstdc++

**Comment by evetion** [comment]
I'd say yes! @visr?

**Comment by visr** [comment]
Great, thanks for working on this! I just opened a few random logs like [this](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=6682&view=logs&j=c9939d36-165e-5893-18bf-a257e243c9cb&t=f84ef0f2-b1ac-5cb4-f5af-f70092a366bd&l=4650) and see what we want to see:

> ZSTD support:              yes
> WebP support:              yes

So this is good to go. @evetion would be great if you could update https://github.com/JuliaGeo/GDAL.jl/blob/ac27d0f35569a560abd813f217d90beaaf1a5f76/test/drivers.jl correspondingly after this is merged, such that we can test regressions in the build.

**Comment by giordano** [comment]
Maybe it'd be good to add checks similar to these: https://github.com/JuliaPackaging/Yggdrasil/blob/14361cc0c37cb6458e08c7fb04449bbae6bc7b4d/G/GDAL/build_tarballs.jl#L52-L55

**Comment by evetion** [comment]
> Maybe it'd be good to add checks similar to these:
> 

Done. And, for good measure, also tried to add a Postgres driver. Let's see how that works out.

**Comment by evetion** [comment]
> So this is good to go. @evetion would be great if you could update https://github.com/JuliaGeo/GDAL.jl/blob/ac27d0f35569a560abd813f217d90beaaf1a5f76/test/drivers.jl correspondingly after this is merged, such that we can test regressions in the build.
Will do after this is merged. 👍

**Comment by evetion** [comment]
Ok, it seems PostgreSQL is not as easy as the other drivers, I'll do that in another PR. It seems the builds failed with it failing to find the header files, such as [here](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=6685&view=logs&j=ab2cc3b2-3c01-5670-75d2-b343be9bfac0&t=97d3e092-fee5-5733-89b3-652a39c01b10&l=48689). Sorry for the noise.

**Comment by evetion** [comment]
@giordano This is ready to merge, only the CI needs a little nudge.
